### PR TITLE
part 7 was implemented

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

File: src/main/java/com/manufacturing/controllers/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
```
Comment (Max-Pochepets) [inline src/main/java/com/manufacturing/controllers/LoginController.java:30]:
```suggestion
        String password = req.getParameter("password");
```
:)

File: src/main/java/com/manufacturing/controllers/driver/CreateDriverController.java [hunk 2]
```diff
@@ -22,11 +22,22 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+        driver.setLogin(login);
+        if (password.equals(passwordRepeat)) {
+            driver.setPassword(password);
+            driverService.create(driver);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } else {
+            req.setAttribute("message", "passwords should be equals");
+            req.setAttribute("log", login);
+            req.getRequestDispatcher("/WEB-INF/views/driver/create.jsp").forward(req, resp);
+        }
```
Comment (Max-Pochepets) [inline src/main/java/com/manufacturing/controllers/driver/CreateDriverController.java]: we don't need to repeat the password in the scope of this task

File: src/main/java/com/manufacturing/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import com.manufacturing.service.DriverService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver byLogin = driverService.findByLogin(login)
```
Comment (Max-Pochepets) [inline src/main/java/com/manufacturing/security/AuthenticationServiceImpl.java]:
```suggestion
        Driver driverByLogin = driverService.findByLogin(login)
```

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String servletPath = req.getServletPath();
+        if (servletPath.equals("/login") || servletPath.equals("/drivers/create")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        if (req.getSession().getAttribute(DRIVER_ID) == null) {
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
```
Comment (Ketch92) [inline src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java]: you can make it work with a single if construction

**Review by Ketch92 — APPROVED** [review]

File: src/main/java/com/manufacturing/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driverByLogin = driverService.findByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect login or password"));
+        if (driverByLogin.getPassword().equals(password)) {
+            return driverByLogin;
+        }
+        throw new AuthenticationException("Incorrect login or password");
```
Comment (RomanDubovskyi) [inline src/main/java/com/manufacturing/security/AuthenticationServiceImpl.java]:
```suggestion
        Optional<Driver> driverByLogin = driverService.findByLogin(login);
        if (driverByLoging.isPresent() && driverByLogin.get().getPassword().equals(password)) {
            return driverByLogin;
        }
        throw new AuthenticationException("Incorrect login or password");
```

File: src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String servletPath = req.getServletPath();
+        if (servletPath.equals("/login") || servletPath.equals("/drivers/create")) {
+            chain.doFilter(req, resp);
+            return;
```
Comment (RomanDubovskyi) [inline src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java]: it's optional but ideally, we can replace it with Set, so this construction will be more readable when the number of URLs grows.

File: src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector = Injector.getInstance("com.manufacturing");
+    private final DriverService driverService = (DriverService)
```
Comment (RomanDubovskyi) [inline src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java]: do we use it anywhere?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector = Injector.getInstance("com.manufacturing");
+    private final Set<String> urls = new HashSet<>();
```
Comment (RomanDubovskyi) [inline src/main/java/com/manufacturing/web/filters/AuthenticationFilter.java]:
```suggestion
    private final Set<String> allowedUrls = new HashSet<>();
```
URLs name is a bit ambigours

**Review by RomanDubovskyi — APPROVED** [review]
Good job 👍

**Pull request merged**